crates/autopilot/src/run_loop.rs

```diff
@@ -239,7 +239,7 @@ impl RunLoop {
         }
 
         let competition_simulation_block = self.eth.current_block().borrow().number;
-        let block_deadline = competition_simulation_block + self.config.submission_deadline;
+        let block_deadline = auction.block + self.config.submission_deadline;
```

Comment:
> Since the `auction block` is often one block behind the `competition_simulation_block`.

For the majority, that shouldn't be the case. If the auction block observation delay exceeds the configured one(which is 2s for mainnet), a new auction doesn't get cut until the next block. So we always ensure we have at least ~10s before the next block to create a new auction, which gets created in ~1-2s. So, there shouldn't be a difference between the auction block and competition simulation block in most cases.

Actually, after writing this comment, I now doubt that the PR changes make sense. At least, we should have better logging to help find the root cause.

Reply:
Sounds great, thank you!

Reply:
Ok, I forgot to include the solving time, so that is expected. The deadline will be increased by 1.